Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -572,10 +572,14 @@ private func checkForUpdates() {
 }
 
 private func checkForLocationUsageDescription() {
+    #if SWIFT_PACKAGE
```

Comment:
This disables the check even during production when using SPM. We need to find some way of only disabling the check when running tests built using SPM. For example, the test target could set a value in NSUserDefaults that this code can check for. The code in the test target that sets the default could be conditioned on `SWIFT_PACKAGE` while the code that checks for the default can check it regardless of the package manager. (Same thing in LegacyRouteController.swift.)

Reply:
Thanks for the suggestion, I'll work on this part

Reply:
I add the `UserDefaults` info for location usage in the `MapboxCoreNavigationTests.swift` cases, and the check of the  `UserDefaults` info check in the `func checkForLocationUsageDescription()` to avoid the` Info.plist` check when running tests built using SPM.